hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableFactory.java

```diff
@@ -86,12 +86,15 @@ public DynamicTableSource createDynamicTableSource(Context context) {
     setupTableOptions(conf.getString(FlinkOptions.PATH), conf);
     ResolvedSchema schema = context.getCatalogTable().getResolvedSchema();
     setupConfOptions(conf, context.getObjectIdentifier(), context.getCatalogTable(), schema);
+    HoodieTableMetaClient metaClient = StreamerUtil.metaClientForReader(conf, HadoopConfigurations.getHadoopConf(conf));
+    SanityCheckUtil.sanitCheck(conf, schema, metaClient);
     return new HoodieTableSource(
         schema,
         path,
         context.getCatalogTable().getPartitionKeys(),
         conf.getString(FlinkOptions.PARTITION_DEFAULT_NAME),
-        conf);
+        conf,
+        metaClient);
```

Comment:
The metaClient initialization is costly but the table source only happens once in Job Graph compile time, let's not reuse here to reduce complexity.

Reply:
> The metaClient initialization is costly but the table source only happens once in Job Graph compile time, let's not reuse here to reduce complexity.

Done.